presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateLongPercentileArrayAggregations.java

```diff
@@ -0,0 +1,150 @@
+    public static final InternalAggregationFunction LONG_APPROXIMATE_PERCENTILE_ARRAY_WEIGHTED_AGGREGATION = new AggregationCompiler().generateAggregationFunction(ApproximateLongPercentileArrayAggregations.class, new ArrayType(BIGINT), ImmutableList.<Type>of(BIGINT, BIGINT, new ArrayType(DOUBLE)));
+
+    private ApproximateLongPercentileArrayAggregations() {}
+
+    @InputFunction
+    public static void input(DigestAndPercentileArrayState state, @NullablePosition @BlockPosition @SqlType(StandardTypes.BIGINT) Block block, @BlockIndex int index, @SqlType("array<double>") Block percentilesArrayBlock)
+    {
+        initializePercentilesArray(state, percentilesArrayBlock);
+        initializeDigest(state);
+
+        if (block.isNull(index)) {
+            return;
+        }
+
+        addRawValueToDigest(state, BIGINT.getLong(block, index));
+    }
+
+    @InputFunction
+    public static void weightedInput(DigestAndPercentileArrayState state, @NullablePosition @BlockPosition @SqlType(StandardTypes.BIGINT) Block block, @BlockIndex int index, @SqlType(StandardTypes.BIGINT) long weight, @SqlType("array<double>") Block percentilesArrayBlock)
+    {
+        initializePercentilesArray(state, percentilesArrayBlock);
+        initializeDigest(state);
+
+        if (block.isNull(index)) {
+            return;
+        }
+
+        addWeightedRawValueToDigest(state, BIGINT.getLong(block, index), weight);
+    }
+
+    @CombineFunction
+    public static void combine(DigestAndPercentileArrayState state, DigestAndPercentileArrayState otherState)
+    {
+        QuantileDigest otherDigest = otherState.getDigest();
+        QuantileDigest digest = state.getDigest();
+
+        if (digest == null) {
+            state.setDigest(otherDigest);
+            state.addMemoryUsage(otherDigest.estimatedInMemorySizeInBytes());
+        }
+        else {
+            state.addMemoryUsage(-digest.estimatedInMemorySizeInBytes());
+            digest.merge(otherDigest);
+            state.addMemoryUsage(digest.estimatedInMemorySizeInBytes());
+        }
+
+        state.setPercentiles(otherState.getPercentiles());
+    }
+
+    @OutputFunction("array<bigint>")
+    public static void output(DigestAndPercentileArrayState state, BlockBuilder out)
+    {
+        QuantileDigest digest = state.getDigest();
+        List<Double> percentiles = state.getPercentiles();
+
+        if (digest == null || digest.getCount() == 0) {
+            out.appendNull();
+            return;
+        }
+
+        BlockBuilder blockBuilder = out.beginBlockEntry();
+
+        for (int i = 0; i < percentiles.size(); i++) {
+            Double percentile = percentiles.get(i);
+            BIGINT.writeLong(blockBuilder, digest.getQuantile(percentile));
+        }
+
+        out.closeEntry();
+    }
+
+    public static void initializePercentilesArray(DigestAndPercentileArrayState state, Block percentilesArrayBlock)
```

Comment:
move these static helpers to a utility class, maybe called:
ApproximatePercentileArrayAggregations